### PR TITLE
Implement ECDSA(secp256k1) keys

### LINKED ISSUE #128: Generate ECDSA key from mnemonic phrase

### LINKED ISSUE #129: Add ECDSA key objects

### PR DESCRIPTION
**Pull request by alittley**
Signed-off-by: Austin Littley <austin.littley@swirldslabs.com>

- resolves #129 
- resolves #128

### CODE REVIEW

File: sdk/main/include/impl/BigNumber.h [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  /**
+   * Destructor
+   */
+  ~BigNumber();
+
+  /**
+   * Don't allow copies or moves
+   */
+  BigNumber(const BigNumber&) = delete;
+  BigNumber(BigNumber&& other) = delete;
```
Comment (rwalworth) [inline sdk/main/include/impl/BigNumber.h:45]: I believe you want to delete the copy assignment operator here as well.. with the copy constructor and copy assignment operator deleted, the move constructor and move assignment operator are implicitly deleted as well

File: sdk/main/include/impl/DerivationPathUtils.h [hunk 2]
```diff
@@ -45,6 +46,15 @@ bool isHardenedChildIndex(uint32_t index);
...
 uint32_t getHardenedIndex(uint32_t index);
 
+/**
+ * Converts unsigned 32 bit int index into big endian byte array (direct implementation of ser32 function from BIP 32)
+ *
+ * @param childIndex the index to convert to a big endian byte array
+ *
+ * @return the big endian byte array representing the child index
+ */
+std::vector<unsigned char> ser32(uint32_t childIndex);
```
[the remark this answers is not in the document]
Comment (alittley) [reply sdk/main/include/impl/DerivationPathUtils.h:55]: I named it `indexToBigEndianArray`, so it's clear from the name what is being converted

File: sdk/main/include/impl/BigNumber.h (hunk 1 above)
Comment (alittley) [reply sdk/main/include/impl/BigNumber.h:46]: I made the deletions match what's done in PublicKey

**Review by rwalworth — APPROVED** [review]
LGTM

**Pull request merged**